AustinKellar/PowerOn-Prototype
# Branch Re-Clicking + Small Refactor

Goal: 
When you click on existing branches, they continue growing up to their maximum size.

Important Files Changed:
- BranchSpawner.cs --> ForestController.cs
- BranchScript.cs
- SapScript.cs

What I changed:
- Renamed BranchSpawner.cs to ForestController.cs (since the script now handles all mouse actions, not only branch spawns)
- I did a small refactor of BranchSpawner.cs (now ForestController.cs) and BranchScript.cs in order to make this change easier.
   - Added named #regions so that we can easily collapse and expand important regions of the code in VS
   - Renamed some variables and functions
   - Separated out the click functionalities into different functions that follow the single purpose principle
   - Got rid of the Direction enum in ForestController.cs. Instead, the tree location of the tree is inferred in BranchScript.cs. Right now the tree location is only needed in 1 place, so the enum is not necessary (a comment is fine) but it should be added back in if we need to access tree location somewhere else later.
- Instead of only being able to click on the trees, I had to add another case in ForestController.cs for handling click on existing branches
- When you click branches that already exist and that aren't grown to max size, they continue growing
- When a branch stops growing, the script doesn't get deleted any more, since it could get called again. Instead it keeps track of whether or not its growing with a boolean property called IsGrowing

How to Test:
- Since there is a refactor, play with the develop branch to gauge its behavior
- Fetch the new branch and test it against develop to make sure it feels the same.
   - The only difference there should be is that I fixed the maximum length of branches. I failed to notice a bug in the PR that added the art where the branches could grow way too long (you can confirm this in dev)
- Try to click on existing branches
   - If you have sap and the branches are not at max length, they are expected to continue growing until they reach max length or you run out of sap

Notes:
- I looked into adding a different mouse cursor when hovering over clickable items (such as branches and tree trunks) but it seems that its not trivial in Unity. It should probably be a separate PR.
